api/passport/facebook.js

```diff
@@ -6,24 +6,34 @@ const UserController = require('../controllers/user');
 const FBStrategy = {
   clientID: process.env.FACEBOOK_APP_ID,
   clientSecret: process.env.FACEBOOK_APP_SECRET,
-  callbackURL: process.env.FACEBOOK_CALLBACK_URL,
   profileFields: ['id', 'emails', 'name', 'displayName', 'gender', 'picture']
 };
 
 passport.use(new FacebookStrategy(FBStrategy, UserController.facebookLogin));
 
 const configureFacebookStrategy = app => {
-  app.get(
-    '/login/facebook',
-    passport.authenticate('facebook', {
+  let domain;
+  app.get('/login/facebook', (req, res, next) => {
+    domain = req.headers.referer;
```

Comment:
I would personally prefer defining `domain`  in both endpoints instead of assigning it a value in the first endpoint and reusing it in the second. Another option would be to write a middleware function that sets the domain on the request object, and then you'd have access to it from the endpoints:

```javascript
// In `app.js`.
app.use(middleware.swaggerUi());

app.use((req, res, next) => {
  req.domain = req.headers.referer || 'https://app.cboard.io/';
  next();
});

Facebook.configureFacebookStrategy(app);
Google.configureGoogleStrategy(app);
```

Reply:
Agree to Simon, please update it @tomivm